include/jemalloc/internal/spin_delay_arm.h

```diff
@@ -0,0 +1,23 @@
+#include "jemalloc/internal/jemalloc_preamble.h"
+#include <stdatomic.h>
+
+/* Global variable to track SB support, declared as extern to be defined in one TU */
+extern _Atomic int arm_has_sb_instruction;
+
+/* Constructor function declaration - implementation in spin_delay_arm.c */
+__attribute__((constructor))
+void detect_arm_sb_support(void);
```

Comment:
This constructor function declaration should not appear in the header anymore. It should only appear in, and only be marked as a constructor in, one TU.